src/seedu/addressbook/Main.java

```diff
@@ -105,17 +108,32 @@ private void recordResult(CommandResult result) {
      */
     private CommandResult executeCommand(Command command)  {
         try {
+        	storage.checkFileStillPresent();
             command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
+            CommandResult result = command.execute(); 
             storage.save(addressBook);
             return result;
+        } catch (FileNotFoundException e) {
+        	return resolveDeletedStorageFile(command, e);
         } catch (Exception e) {
             ui.showToUser(e.getMessage());
             throw new RuntimeException(e);
         }
     }
 
-    /**
+	private CommandResult resolveDeletedStorageFile(Command command, FileNotFoundException errorMessage) {
+		try {
+			ui.showToUser(errorMessage.getMessage());
+			storage.recreateStorageFile(addressBook);
```

Comment:
re creating the file is done by the code already, you don't need to do it explicitly